pkg/controller/gcp/cache/claim.go

```diff
@@ -73,26 +72,34 @@ func ConfigureCloudMemorystoreInstance(_ context.Context, cm resource.Claim, cs
 		return errors.Errorf("expected resource claim %s to be %s", cm.GetName(), cachev1alpha1.RedisClusterGroupVersionKind)
 	}
 
-	rs, csok := cs.(*corev1alpha1.ResourceClass)
+	rl, csok := cs.(*v1alpha1.CloudMemorystoreInstanceClass)
 	if !csok {
-		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), corev1alpha1.ResourceClassGroupVersionKind)
+		return errors.Errorf("expected resource class %s to be %s", cs.GetName(), v1alpha1.CloudMemorystoreInstanceClassGroupVersionKind)
 	}
 
 	i, mgok := mg.(*v1alpha1.CloudMemorystoreInstance)
 	if !mgok {
 		return errors.Errorf("expected managed resource %s to be %s", mg.GetName(), v1alpha1.CloudMemorystoreInstanceGroupVersionKind)
 	}
 
-	spec := v1alpha1.NewCloudMemorystoreInstanceSpec(rs.Parameters)
+	// NOTE(hasheddan): Getting rid of the NewCloudMemorystoreInstanceSpec func creates a dependency on corev1alpha1.
+	// This seems fine to me. We are already depending on cachev1alpha1.
+	spec := &v1alpha1.CloudMemorystoreInstanceSpec{
+		ResourceSpec: corev1alpha1.ResourceSpec{
+			ReclaimPolicy: corev1alpha1.ReclaimRetain,
+		},
+		CloudMemorystoreInstanceParameters: rl.SpecTemplate.CloudMemorystoreInstanceParameters,
```

Comment:
yes, this is so nice basically just copying over the strongly typed fields from the class, instead of having to parse and convert all of them like we used to in `NewCloudMemorystoreInstanceSpec`.  great!